src/extractor/source.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Source of data to be extracted, like a DOM document, a URL or any other kind of resource.
+ * For the moment, only DOM Document is supported.
```

Comment:
Could you elaborate on these other kinds of resources? The URL can be accessed through `document.location.href`.

Reply:
The idea behind having a `Source` interface is so that the API does not depend on a specific data structure, that might not be available in all runtimes. If in the future we would like to run an extractor in `nodejs`, for example, the `document` would not exist (or a least would not have the same type).

Another reason would be that we can envision having extractors that don't rely on a `document`, but instead, for example, pull directly from a URL. (We could also make it so that an extractor can support multiple types of `Sources`, e.g. `DOMSource` and `URLSource`).

If we would not introduce the notion of a `Source` at this moment, adding support later for multiple types of sources would be a breaking change to the API, which would require updating all existing extractors.